server/lib/sanitize-html.ts

```diff
@@ -140,5 +140,15 @@ export const generateSummaryForHTML = (content, maxLength = 255) => {
   const truncated = truncate(trimmed, { length: maxLength });
 
   // Second sanitize pass: an additional precaution in case someones finds a way to play with the trimmed version
-  return sanitizeHTML(truncated, optsSanitizeSummary);
+  const secondSanitized = sanitizeHTML(truncated, optsSanitizeSummary);
+
+  const isTruncated = trimmed.length > maxLength;
+  const fullySanitized = sanitizeHTML(secondSanitized);
```

Comment:
Sanitizing an already-sanitized string won't have any effect. It's useful after truncating because it will clean invalid markup, but here we're basically doing `sanitizeHTML(sanitizeHTML(truncated, optsSanitizeSummary))`.

Also we must not forget to pass `optsSanitizeSummary` to the function when we use it.

Reply:
The only reason I am using that final sanitize (`const fullySanitized = sanitizeHTML(secondSanitized)`) is to see if it contains the `...` in the if statement below that. Since we don't pass the `optsSanitizeSummary` this will remove all the html tags and thus will give us whether or not it's actually truncated or not. Do you have a better idea to handle this case? :thinking:

Reply:
I get the idea, I missed that during my first review. The `...` should normally always be outside of tags, we don't want them to be included in whatever styling (bold, strikethrough) that may be occurring here.

So I'd say it should be safe to test `secondSanitized` directly:
```es6
const isTruncated = trimmed.length > maxLength;
const secondSanitized = sanitizeHTML(truncated, optsSanitizeSummary);

// Check to see if the second sanitization cuts a html tag in the middle
if (isTruncated && secondSanitized.slice(secondSanitized.length - 3) !== '...') {
  return `${secondSanitized.trim()}...`;
} else {
   return secondSanitized;
}
```

Reply:
> I get the idea, I missed that during my first review. The `...` should normally always be outside of tags, we don't want them to be included in whatever styling (bold, strikethrough) that may be occurring here.
> 
> So I'd say it should be safe to test `secondSanitized` directly:
> 
> ```js
> const isTruncated = trimmed.length > maxLength;
> const secondSanitized = sanitizeHTML(truncated, optsSanitizeSummary);
> 
> // Check to see if the second sanitization cuts a html tag in the middle
> if (isTruncated && secondSanitized.slice(secondSanitized.length - 3) !== '...') {
>   return `${secondSanitized.trim()}...`;
> } else {
>    return secondSanitized;
> }
> ```

@Betree : This will fail the test [`Doesn't cut anchors`](https://github.com/opencollective/opencollective-api/blob/398edc15b23ec27d6ad37c013c8b4c6e37f1a3f8/test/server/lib/sanitize-html.test.js#L142). Because the if condition will pass and we'll add an additional set of `...` at the end. Am I missing something? :thinking: :thinking:

Reply:
It's ok to update the test, in this example the `...` should actually be outside of the `strong` because we don't want them to be formatted. 

```es6
expect(generateSummaryForHTML("I'd like to say <strong>Hello World</strong>", 30)).to.to.eq(
  "I'd like to say <strong>Hel</strong>...",
);```